platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/PropertyOperation.java

```diff
@@ -0,0 +1,203 @@
+
+    final Map<String, String> currentProperties;
+
+    private PropertyOperation(Path configFile, Map<String, String> configs,
+            boolean loadCurrentProps) {
+        this.configFile = configFile.toFile();
+        this.configs = new HashMap<>(configs);
+
+        currentProperties = new HashMap<>();
+        if (loadCurrentProps) {
+            Properties result;
+            try {
+                Properties props = new Properties();
+                try (FileInputStream in = new FileInputStream(configFile.toFile())) {
+                    props.load(in);
+                }
+
+                result = props;
+            } catch (IOException e) {
+                throw new ConfiguratorException(String.format(
+                        "Error reading configuration from file %s",
+                        configFile.toFile()
+                                .getName()));
+            }
+
+            result.keySet()
+                    .stream()
+                    .map(String.class::cast)
+                    .forEach(k -> currentProperties.put(k, result.getProperty(k)));
+        }
+    }
+
+    /**
+     * Creates a handler for persisting property file changes to a new property file.
+     *
+     * @param configFile the property file to be created
+     * @param configs    map of key:value pairs to be written to the property file
+     * @return instance of this class
+     */
+    public static PropertyOperation forCreate(Path configFile, Map<String, String> configs) {
+        return new PropertyOperation.CreateHandler(configFile, configs);
+    }
+
+    /**
+     * Creates a handler for deleting a property file.
+     *
+     * @param configFile the property file to be deleted
+     * @return instance of this class
+     */
+    public static Operation forDelete(Path configFile) {
+        return new PropertyOperation.DeleteHandler(configFile);
+    }
+
+    /**
+     * Creates a handler for persisting property file changes to an existing property file.
+     *
+     * @param configFile  the property file to be updated
+     * @param configs     map of key:value pairs to be written to the property file
+     * @param keepIgnored if true, any keys in the current property file that are not in the
+     *                    {@code configs} map will be left with their initial values; if false, they
+     *                    will be removed from the file
+     * @return instance of this class
+     */
+    public static PropertyOperation forUpdate(Path configFile, Map<String, String> configs,
+            boolean keepIgnored) {
+        return new PropertyOperation.UpdateHandler(configFile, configs, keepIgnored);
+    }
+
+    @Override
+    public Void rollback() throws ConfiguratorException {
+        saveProperties(currentProperties);
+
+        return null;
+    }
+
+    @Override
+    public Map<String, String> readState() throws ConfiguratorException {
+        return currentProperties;
```

Comment:
Isn't it dangerous to return a reference to the internal maps? Do we account for outside change to it?